### ISSUE TITLE
Inconsistent PreTrainedTokenizerBase.pad  argument default value & docstring

### ISSUE THREAD

**Issue by Fraser-Greenlee** [huggingface/transformers#8837]
The docstring states the argument `padding` has a default of `False` but its default is `True`

docstring: 
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2469-L2470
arg:
https://github.com/huggingface/transformers/blob/d5b3e56de5376aa85ef46e7f0325139d9e299a41/src/transformers/tokenization_utils_base.py#L2431-L2472

This causes issues when using `DataCollatorForLanguageModeling` with an already padded dataset as it resets the attention mask.

**Comment by Fraser-Greenlee** [comment]
Seems to have been added in this commit: 
https://github.com/huggingface/transformers/commit/f3065abdb8805f5beaed9ff1e92ce874e655f5c9#diff-85b29486a884f445b1014a26fecfb189141f2e6b09f4ae701ee758a754fddcc1R2146-R2168
As part of merge https://github.com/huggingface/transformers/pull/6110

**Comment by LysandreJik** [comment]
Hi, indeed! The docs should be changed to reflect the method signature. Do you want to open a PR?
